osbuild/objectstore.py

```diff
@@ -180,27 +186,15 @@ def finalize(self):
         self._mode = Object.Mode.READ
 
     def cleanup(self):
-        workdir = self._workdir
-        if workdir:
-            # manually remove the tree, it might contain
-            # files with immutable flag set, which will
-            # throw off standard Python 3 tempdir cleanup
-            rmrf.rmtree(os.path.join(workdir.name, "data"))
-
-            workdir.cleanup()
-            self._workdir = None
+        if self._stack:
+            self._stack.close()
+            self._stack = None
```

Comment:
`__exit__()` does this, right? What is the reason we need `cleanup()`? And then, maybe call `cleanup()` from `__exit__()` and make sure to reset the variables that depend on the values in the exitstack like `_path`?

Reply:
Actually, yes. I do prefer having everything in `cleanup()` and then calling that from `__exit__`, not sure why I didn't stick to it here. I will change it.